### PR TITLE
#10 [feat/solana-nft] NFT buy/sell using Tensor SDK/API

### LINKED ISSUE #10: [feat/solana-nft] NFT buy/sell using Tensor
# Overview
Using Tensor API for buying NFT from Tensor Marketplace

# Feature

- [ ] Buying the lowest listing NFT from a particular collection(e.g: I want to buy a Drip Nation NFT on Tensor using my available balance) 
- [ ] Buying limit orders(e.g: Can you purchase me 1 Okay Bears NFT for 10 SOL ?.)
- [ ] Buying a set of NFT in bulk when they satisfy specific criteria.
# Resources: 

https://tensor-hq.notion.site/PUBLIC-Tensor-Trade-API-Docs-alpha-b18e1a196187473bac9b5d6de5b47032

note for @batm9n: Could you please confirm if I am heading in the right track? Because it seems that in order to construct a buy tx from user intent, we need to somehow retrieve the onchainId of the NFT, the marketplace name(e.g: Tensor). But copy the id seem unintuitive, and I think it is more feasible to choose NFT for user based on some criteria(for now I think lowest price is easy to implement)

### PR DESCRIPTION
**Pull request by TropicalDog17**
#10 

Tested query: Can you purchase me 1 DegenPoet NFT? (choose this because reasonable price, has many listing so don't need to worry about run out of nft to test :sunglasses:  )

Can add support for more NFT collections by providing their info on constant.js file

### CODE REVIEW

**Comment by TropicalDog17** [comment]
Not sure if Helius RPC help solve the problem of sending NFT, or maybe I am just blocked by sending too much reqs to the mainnet-beta RPC

anyway, get one Helius RPC endpoint here: https://www.helius.dev

From https://docs.solana.com/cluster/rpc-endpoints
``` The public RPC endpoints are not intended for production applications. Please use dedicated/private RPC servers when you launch your application, drop NFTs, etc. The public services are subject to abuse and rate limits may change without prior notice. Likewise, high-traffic websites may be blocked without prior notice. ```

https://solana.stackexchange.com/questions/4576/on-browser-mainnet-beta-rpc-error-403-access-forbidden-but-works-in-python

**Comment by MrSufferer** [comment]
@TropicalDog17 
Yeah, you shouldn't use the official RPC url for mainnet-beta.

Get a new one(from Alchemy or Helius).
Then, place it at the `REACT_APP_SOLANA_RPC_URL` within .env file.

**Comment by TropicalDog17** [comment]
@batm9n Have you tested buying a NFT yet? 
And maybe we could refactor so that any devnet-compatible operation will operate in devnet cluster, and the others will go through mainnet-beta cluster, either officially from Solana or from dedicated RPCs(Helius/Alchemy)

**Comment by MrSufferer** [comment]
@TropicalDog17

**I tried to sign the message 1st time:**

Screen Shot 2023-10-04 at 21 21 11

But on the 2nd and 3rd try, it was okay (_already tested on a different wallet address_).
Seems like the problem originated from my Backpack wallet.


### About the devnet/mainnet categorization:

I agree with this, we should add a switch button that changes the network context state.

File: handler/.env.sample
Comment (MrSufferer) [inline handler/.env.sample:1]: Lack an ENDLINE

File: handler/utils/tensorQuery.js
Comment (MrSufferer) [inline handler/utils/tensorQuery.js:1]: lack an endline

File: handler/apitest.js
Comment (MrSufferer) [inline handler/apitest.js:1]: line 24: put query text down to line 25

File: handler/constants.js
Comment (MrSufferer) [inline handler/constants.js:1]: line 300, line 322

**Comment by MrSufferer** [comment]
@TropicalDog17 Nice work!

**Pull request merged**